swarm/netsim/simulation.go

```diff
@@ -109,12 +96,11 @@ func New(services map[string]ServiceFunc, opts *SimulationOptions) (s *Simulatio
 		&simulations.NetworkConfig{ID: "0"},
 	)
 
-	if opts == nil {
-		opts = new(SimulationOptions)
-	}
-	if opts.WithHTTP {
-		s.initHTTPServer(opts)
-	}
+	return s
+}
+
+func (s *Simulation) WithServer(ctx context.Context, port string) *Simulation {
+	s.startHTTPServer(port)
```

Comment:
Doesn't this leak the HTTP server? I think it should be shutdown when the context is cancelled.

Reply:
Context is not needed here. Actually, maybe it is better just to remove startHTTPServer and have only WithServer function in http.go file.

The server is shut down on Simulation.Close https://github.com/holisticode/go-ethereum/blob/05729f4e73399e77575765f49c3a7bb440119d6d/swarm/netsim/simulation.go#L191.